### PR TITLE
Refactor alertsManager.ts

### PR DESCRIPTION
**Pull request by lucasponce**
This is a preliminar refactor of alertsManager.ts in order to unify how Alerts API is being used from the UI.
We need to upgrade to Alerts 0.4.x and that has changes on the ui-services, but before to upgrade I think is good trying to simplify and use a consistent way to create triggers and fetch alerts.
There are some parts commented in datasources code as it will be simplify with 0.4.x when we can modify conditions in a single call.
@mtho11 and @jshaughn please, can you take a look ?
If this is ok, then tomorrow we can work upgrading 0.4.x + ui-services and fixing missing things on UI code.

### CODE REVIEW

File: console/src/main/scripts/plugins/metrics/ts/app-details/appServerDatasourcesAlerts.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ module HawkularMetrics {
         connAlertDefinition.conditions[0].threshold = this.adm.conn.conditionThreshold;
       }
 
-      var connSavePromise = this.HawkularAlertsManager.saveAlertDefinition(connAlertDefinition,
+      var connSavePromise = this.HawkularAlertsManager.updateTrigger(connAlertDefinition,
```
Comment (jshaughn) [inline console/src/main/scripts/plugins/metrics/ts/app-details/appServerDatasourcesAlerts.ts]: If you are updating terminology I would also change the variable name.

File: console/src/main/scripts/plugins/metrics/ts/app-details/appServerDatasourcesAlerts.ts [hunk 2]
```diff
@@ -170,7 +191,7 @@ module HawkularMetrics {
         respAlertDefinition.conditions[idCreation].threshold = this.adm.resp.creationTimeThreshold;
       }
 
-      var respSavePromise = this.HawkularAlertsManager.saveAlertDefinition(respAlertDefinition,
+      var respSavePromise = this.HawkularAlertsManager.updateTrigger(respAlertDefinition,
```
Comment (jshaughn) [inline console/src/main/scripts/plugins/metrics/ts/app-details/appServerDatasourcesAlerts.ts]: same, possibly change var name.

File: console/src/main/scripts/plugins/metrics/ts/metricsAlerts.ts [hunk 3]
```diff
@@ -251,7 +251,7 @@ module HawkularMetrics {
       availabilityAlertDefinition.dampenings[0].evalTimeSetting = this.adm.avail.responseDuration;
       availabilityAlertDefinition.dampenings[1].evalTimeSetting = this.adm.avail.responseDuration;
 
-      var availabilitySavePromise = this.HawkularAlertsManager.saveAlertDefinition(availabilityAlertDefinition,
+      var availabilitySavePromise = this.HawkularAlertsManager.updateTrigger(availabilityAlertDefinition,
```
Comment (jshaughn) [inline console/src/main/scripts/plugins/metrics/ts/metricsAlerts.ts]: var name again

File: console/src/main/scripts/plugins/metrics/ts/metricsAlerts.ts [hunk 4]
```diff
@@ -292,7 +292,7 @@ module HawkularMetrics {
         responseAlertDefinition.conditions[1].threshold = this.adm.thres.conditionThreshold;
       }
 
-      var responseSavePromise = this.HawkularAlertsManager.saveAlertDefinition(responseAlertDefinition,
+      var responseSavePromise = this.HawkularAlertsManager.updateTrigger(responseAlertDefinition,
```
Comment (jshaughn) [inline console/src/main/scripts/plugins/metrics/ts/metricsAlerts.ts]: again

**Comment by mtho11** [comment]
Looks like it needs a rebase due to global commit that went in earlier.

**Comment by jshaughn** [comment]
Other than the variable names, this looks OK to me.

File: console/src/main/scripts/plugins/metrics/ts/app-details/appServerDatasourcesAlerts.ts [hunk 5]
```diff
@@ -28,7 +28,7 @@ module HawkularMetrics {
 
       var connTriggerId = this.resourceId + '_ds_conn';
 
-      var connDefinitionPromise = this.HawkularAlertsManager.getAlertDefinition(connTriggerId)
+      var connDefinitionPromise = this.HawkularAlertsManager.getTrigger(connTriggerId)
```
Comment (mtho11) [inline console/src/main/scripts/plugins/metrics/ts/app-details/appServerDatasourcesAlerts.ts]: Please change var to ES6 let. *Let* is the new *var*.

**Comment by mtho11** [comment]
LGTM Lucas. Ship it! ;)

**Comment by lucasponce** [comment]
@mtho11, @jshaughn, I've sent a feedback commit and rebase with master, can you check everything looks good ?

**Comment by mtho11** [comment]
Maarvelous!

**Pull request merged**